slack-moderator/handler.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"sigs.k8s.io/slack-infra/slack"
+)
+
+type handler struct {
+	client     *slack.Client
+	adminToken string
+}
+
+// ServeHTTP handles Slack webhook requests.
+func (h *handler) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
```

Comment:
this function seems like it should probably be broken out into some common subroutine at some point, at least parts of it

Reply:
(also non-blocking, this method is fine, just seems like most of it would already be re-usable if pulled out somewhere)